src/test/groovy/org/gradle/profiler/ScenarioLoaderTest.groovy

```diff
@@ -248,6 +249,39 @@ class ScenarioLoaderTest extends Specification {
         "cli"         | "none" | 1       | 1
     }
 
+    def "uses warm-up and iteration counts defined by scenario"() {
+        def benchmarkSettings = settings()
+        def profileSettings = settings(GradleBuildInvoker.ToolingApi, false)
+
+        scenarioFile << """
+            default {
+                run-using = ${runUsing}
+                daemon = ${daemon}
+                warm-ups = 5
+                iterations = 2
+            }
+        """
+        def benchmarkScenarios = loadScenarios(scenarioFile, benchmarkSettings, Mock(GradleBuildConfigurationReader))
+        def profileScenarios = loadScenarios(scenarioFile, profileSettings, Mock(GradleBuildConfigurationReader))
+
+        expect:
+        def benchmarkScenario = benchmarkScenarios[0] as GradleScenarioDefinition
+        benchmarkScenario.warmUpCount == 5
+        benchmarkScenario.buildCount == 2
+
+        def profileScenario = profileScenarios[0] as GradleScenarioDefinition
+        profileScenario.warmUpCount == 5
+        profileScenario.buildCount == 2
+
+        where:
+        runUsing      | daemon | warmups | profileWarmups
```

Comment:
I don't see these parameters referenced, but they should be, right?

Reply:
Removed `warmups` and `profileWarmups`